x-pack/plugins/ui_actions_enhanced/public/custom_time_range_badge.tsx

```diff
@@ -52,12 +52,13 @@ export class CustomTimeRangeBadge implements Action<TimeBadgeActionContext> {
   }
 
   public getDisplayName({ embeddable }: TimeBadgeActionContext) {
-    return prettyDuration(
-      embeddable.getInput().timeRange.from,
-      embeddable.getInput().timeRange.to,
-      commonDurationRanges,
-      this.dateFormat
-    );
+    return (
+      <PrettyDuration
+        timeFrom={embeddable.getInput().timeRange.from}
+        timeTo={embeddable.getInput().timeRange.to}
+        dateFormat={this.dateFormat}
+      />
+    ) as unknown as string; // This is not a string, but this is used in EuiBadge which accepts JSX
```

Comment:
@elastic/kibana-app-services if there's a better way of overriding `getDisplayName`'s returned type as a one-off, please let me know. `getDisplayName` in this case is used within an `EuiBadge` (screenshot below) which accepts JSX as well as a string.

<img width="467" alt="" src="https://user-images.githubusercontent.com/549407/159804969-28cba0bc-e66f-4937-919d-af4df307af1e.png">

Reply:
Maybe better for safety and to avoid incorrect type casting, let's convert this to string? 

For example like here 

https://github.com/elastic/kibana/blob/a5620502d0d9aefc87e07ddeb47fe7ae2c539f3b/src/plugins/data_view_field_editor/public/components/preview/field_preview_context.tsx#L51

Reply:
https://github.com/elastic/kibana/pull/128825/commits/baf57143a6760e0f3e7832de259d38eeb8a4efbe - Awesome, this is still working locally as expected using `renderToString`. Thanks so much for learning me a handy new React util! 🙌

<img width="304" alt="" src="https://user-images.githubusercontent.com/549407/161102115-fc09a2cb-c2f7-48b1-a6c4-f79bc1aeaa49.png">